src/events.jl

```diff
@@ -35,19 +35,15 @@ end
 
 mutable struct Gtk_signal_motion{T}
     closure::T
-    callback::Ptr{Nothing}
+    callback::Base.CFunction
     include::UInt32
     exclude::UInt32
 end
 function notify_motion(p::Ptr{GObject}, eventp::Ptr{GdkEventMotion}, closure::Gtk_signal_motion{T}) where T
     event = unsafe_load(eventp)
     if event.state & closure.include == closure.include &&
        event.state & closure.exclude == 0
-        if isbitstype(T)
-            ret = ccall(closure.callback, Cint, (Ptr{GObject}, Ptr{GdkEventMotion}, T), p, eventp, closure.closure)
-        else
-            ret = ccall(closure.callback, Cint, (Ptr{GObject}, Ptr{GdkEventMotion}, Any), p, eventp, closure.closure)
-        end
+        ret = closure.callback.f(p, eventp, closure.closure)
```

Comment:
```suggestion
        ret = closure.callback(p, eventp, closure.closure)
```

Reply:
This would be preferable to avoid increasing yet more the dependence on internal design details.

Reply:
I agree. I don't think we have a call overload for `CFunction` though.

Reply:
No, but let's just not allocate it then

Reply:
Oh, wait a minute, is this only called from julia? Why was it using `cfunction` in the first place? @tknopp any reason?